### ISSUE TITLE
Cannot create new page on 2.1.0-beta.1

### ISSUE THREAD

**Issue by manucorporat** [driftyco/ionic-cli#1381]
Reported by @athosbr99 in https://github.com/driftyco/ionic/issues/8104 
>On 2.1.0-beta.1, when created a new page using ionic g page <namepage>, a ENOENT appears.
I uninstalled and installed the stable ionic package and this error doesn't occur. I uninstalled and installed ionic@beta again and it happened again. I created a new project and the issue persists.

```
Error: ENOENT: no such file or directory, scandir '/home/athos/Git/if55a/Lista3/agenda/node_modules/ionic-angular/templates/page'
    at Error (native)
    at Object.fs.readdirSync (fs.js:808:18)
    at getTemplatesInDir (/usr/local/lib/node_modules/ionic/node_modules/@ionic/app-generators/index.js:23931:65)
    at Generator.renderAndWriteTemplates (/usr/local/lib/node_modules/ionic/node_modules/@ionic/app-generators/index.js:25309:116)
    at /usr/local/lib/node_modules/ionic/node_modules/@ionic/app-generators/index.js:25304:26


ENOENT: no such file or directory, scandir '/home/athos/Git/if55a/Lista3/agenda/node_modules/ionic-angular/templates/page' (CLI v2.1.0-beta.1)

```

**Run `ionic info` from terminal/cmd prompt:** (paste output below)

Cordova CLI: 6.3.1
Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.1.0-beta.1
Ionic App Lib Version: 2.1.0-beta.0
OS: Distributor ID:	Ubuntu Description:	Ubuntu 16.04.1 LTS 
Node Version: v4.2.6

**Comment by biinio** [comment]
Hello Ionic,
I'm having the same issue after trying to run:

ionic g page TestProvider

I get the error:

Error: ENOENT: no such file or directory, scandir '/Users/Esteban/Development/ionic2-providers/node_modules/ionic-angular/templates/page'
    at Error (native)
    at Object.fs.readdirSync (fs.js:951:18)
    at getTemplatesInDir (/usr/local/lib/node_modules/ionic/node_modules/@ionic/app-generators/index.js:23931:65)
    at Generator.renderAndWriteTemplates (/usr/local/lib/node_modules/ionic/node_modules/@ionic/app-generators/index.js:25309:116)
    at /usr/local/lib/node_modules/ionic/node_modules/@ionic/app-generators/index.js:25304:26


ENOENT: no such file or directory, scandir '/Users/Esteban/Development/ionic2-providers/node_modules/ionic-angular/templates/page' (CLI v2.1.0-beta.1)

Your system information:

Cordova CLI: 6.3.1
Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.1.0-beta.1
Ionic App Lib Version: 2.1.0-beta.0
ios-deploy version: 1.9.0 
ios-sim version: 5.0.8 
OS: Mac OS X El Capitan
Node Version: v6.5.0
Xcode version: Xcode 8.0 Build version 8A218a 


I thought you should now.
Thank you

**Comment by rabe77** [comment]
Hello, i had the same issues on Windows 7 SP1, and Windows 8.1 computers,

Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.1.0-beta.1
Ionic App Lib Version: 2.1.0-beta.0
OS: Windows 7 SP1
Node Version: v4.5.0

it worked again just fine after downgrading CLI: $ npm install ionic@2.0.0 -g

Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.0.0
Ionic App Lib Version: 2.0.0-beta.20
OS: Windows 7 SP1
Node Version: v4.5.0

No solution, but I also thought that you should know
greetings

**Comment by athosbr99** [comment]
The only workaround so far (I guess) is to rollback to 2.0.0.

**Comment by rcjsuen** [comment]
Hm, it works for me on Windows 10. My CLI version is listed as 2.0.0-beta.37 though.

```
>ionic g page test
√ Create app\pages\test\test.html
√ Create app\pages\test\test.scss
√ Create app\pages\test\test.ts

Don't forget to add an import for test.scss in app\themes\app.core.scss:

  @import "..\pages\test\test.scss";

>ionic info

Your system information:

Cordova CLI: 6.3.1
Gulp version:  CLI version 3.9.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.0.0-beta.37
Ionic App Lib Version: 2.0.0-beta.20
OS:
Node Version: v4.2.2
```

**Comment by Iyashu5040** [comment]
I get the same error on CLI version 2.1.0-beta.2

**Comment by luisravila** [comment]
I get the same error!!!! 

Cordova CLI: 6.3.1
Ionic Framework Version: 2.0.0-beta.11
**Ionic CLI Version: 2.1.0-beta.2**
Ionic App Lib Version: 2.1.0-beta.1
OS: Windows 7 SP1
Node Version: v6.5.0

**Comment by vincenthfrance** [comment]
same error

Cordova CLI: 6.3.1
Gulp version:  CLI version 3.9.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.1.0-beta.2
Ionic App Lib Version: 2.1.0-beta.1
ios-deploy version: 1.9.0 
ios-sim version: 5.0.8 
OS: Mac OS X El Capitan
Node Version: v4.5.0
Xcode version: Xcode 8.0 Build version 8A218a

**Comment by ramonornela** [comment]
Close issue related #8178 @danbucholtz fix in commit https://github.com/driftyco/ionic/commit/60363b29e9630217ed7611bcbdf060dd86ac7dac

**Comment by danbucholtz** [comment]
Hi, sorry, we accidentally released this early. Definitely 110% our bad.

Thanks,
Dan

**Comment by DevlprArup** [comment]
Please fix this issue :

**Comment by DevlprArup** [comment]
I am having the same issue as :

>ionic g page test
--------------------------------

Error: ENOENT: no such file or directory, scandir '...\node_modules\ionic-angular\templates\page'
    at Error (native)
    at Object.fs.readdirSync (fs.js:856:18)
    at getTemplatesInDir (..\AppData\Roaming\npm\node_modules\ionic\node_modules\@ionic\app-generators\index.js:23931:65)
    at Generator.renderAndWriteTemplates (..\AppData\Roaming\npm\node_modules\ionic\node_modules\@ionic\app-generators\index.js:25309:
116)
    at ..\AppData\Roaming\npm\node_modules\ionic\node_modules\@ionic\app-generators\index.js:25304:26


ENOENT: no such file or directory, scandir '..\node_modules\ionic-angular\templates\page' (CLI v2.1.4)

Your system information:

Cordova CLI: 6.3.1
Gulp version:  CLI version 3.9.0
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.1.4
Ionic App Lib Version: 2.1.2
OS: Windows 7 SP1
Node Version: v5.1.0


Please fix this issue ..

**Comment by Iyashu5040** [comment]
@DevlprArup Ionic 2 is on RC1 by now and this issue has been resolved in the latest version. I suggest upgrading your project to RC1. It's quite a process though so make sure to follow the upgrade steps!

**Comment by DevlprArup** [comment]
@Iyashu5040 Thank you so much for your prompt suggestion. Could you please let me know what is the process involved to upgrade my project ? is copying the pages folder to new set up work ? OR do I have to create each page and copy the code needed ? Please help.

**Comment by Iyashu5040** [comment]
@DevlprArup Here's a link to the changelog. This explains a bunch of changes, make sure to look at the changes to how button stylings work now. There's also an explanation of the upgrade procedure there. https://github.com/driftyco/ionic/blob/master/CHANGELOG.md#200-rc0-2016-09-28

**Comment by ramytamer** [comment]
I know it might be a bit old but i fixed this with specifing the path for generators
ex:
`ionic generate page Aboutus --templateDir node_modules/ionic-angular/tooling/generators`

**Comment by kalimulhaq** [comment]
I'm facing same issue in ionic 4

Generators are only available for Ionic 2 projects (CLI v2.2.2)

Your system information:

 ordova CLI: 6.5.0
Ionic Framework Version: 3.0.1
Ionic CLI Version: 2.2.2
Ionic App Lib Version: 2.2.1
Ionic App Scripts Version: 1.3.0
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Windows 10
Node Version: v6.10.2
Xcode version: Not installed
